### ISSUE TITLE
Would you expect it to be a problem that Python 2.7 (Python x,y) is also installed? Is there a particular place that the pymedphys folder must reside?

### ISSUE THREAD

**Issue by SimonBiggs** [pymedphys/pymedphys#10]
In GitLab by @SimonBiggs on Dec 4, 2018, 19:45

@kingrpaul 

https://gitlab.com/pymedphys/pymedphys/merge_requests/86#note_121997468

**Comment by SimonBiggs** [comment]
In GitLab by @SimonBiggs on Dec 4, 2018, 19:46

Yes I do expect that would be an issue. You would need to have a clear way
to define which environment to use.

Personally, if I was in your shoes I would attempt to use 2to3 (https://docs.python.org/2/library/2to3.html) to convert all Python 2 code to Python 3 code, especially given Python 2's only having 13 months left until end of life (https://python3statement.org/ + https://www.python.org/dev/peps/pep-0373/#maintenance-releases).

If that wasn't an option (which would seem odd to me) I would still uninstall all other versions of Python except Anaconda, make sure Anaconda was in the PATH, and only Anaconda Python was installed. Then if I did still need Python 2 I would use Anaconda to manage that (https://conda.io/docs/user-guide/tasks/manage-python.html#installing-a-different-version-of-python).

To minimise confusion there should only be one `pip.exe`, and `python.exe` that can ever be found via your PATH variable without explicitly declaring otherwise with something like conda's activate command (https://conda.io/docs/user-guide/tasks/manage-environments.html#activating-an-environment).

The pymedphys folder doesn't need to reside anywhere special. It just needs to be "unmoving" after `pip install -e .` is run. Only the python that is referred to by that particular `pip` will be able to access the `pymedphys` library. As such having multiple unmanaged python installs can be a source of significant confusion.

**Comment by SimonBiggs** [comment]
In GitLab by @SimonBiggs on Dec 4, 2018, 19:47

mentioned in merge request !86

**Comment by SimonBiggs** [comment]
In GitLab by @centrus on Dec 4, 2018, 22:33

@SimonBiggs, do you think it's worth outlining a recommended use of Anaconda in the docs, or perhaps linking to a useful tutorial? I suspect many physicists would be unfamiliar with managing python versions and packages in conda environments...

**Comment by SimonBiggs** [comment]
In GitLab by @kingrpaul on Dec 5, 2018, 04:05

This is my first encounter with Anaconda and am still having challenges. Now getting authentication failure at the command line when trying to push.

**Comment by SimonBiggs** [comment]
In GitLab by @SimonBiggs on Dec 5, 2018, 05:54

@kingrpaul regarding the authentication failure I've created another issue over at https://gitlab.com/pymedphys/pymedphys/issues/11

That's actually a git + SSH key issue and not an issue with Anaconda.

@centrus yes I certainly agree. I imagine there being quite a few many more docs under the developers section that outlines recommendations and provides links to tutorials. Whichever of us gets a chance first could create a new branch + merge request to add a few more pages to the docs.

I was just sitting down with my supervisor at work and he was having a few environment issues. It was with git (not Anaconda) and it was due to him having an old conflicting version laying around. The solution was to uninstall all versions of git, reset the computer, then reinstall git with choco. I think within the contributor docs for both Python and git we need to recommend that only one version of each resides on the computer, from there we can link to another one of our documents describing how multiple python environments can be set up with conda.
